src/ports/STM32Cube/pnal.c

```diff
@@ -0,0 +1,302 @@
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2021 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "pnal.h"
+
+#include "osal.h"
+#include "osal_log.h"
+#include "options.h"
+
+#include <fatfs.h>
+#include <lwip/apps/snmp.h>
+#include <lwip/netif.h>
+#include <lwip/snmp.h>
+#include <lwip/sys.h>
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int pnal_set_ip_suite (
+   const char * interface_name,
+   const pnal_ipaddr_t * p_ipaddr,
+   const pnal_ipaddr_t * p_netmask,
+   const pnal_ipaddr_t * p_gw,
+   const char * hostname,
+   bool permanent)
+{
+   ip_addr_t ip_addr;
```

Comment:
No checking of correct interface_name?

Reply:
This was copied from rt-kernel/pnal.c. Will leave unchanged for now.